### PR TITLE
Fix documentation for container algorithms

### PR DESCRIPTION
**Pull request by dimitraka**
Fix existing and add missing documentation for ranges

### CODE REVIEW

File: libs/core/algorithms/include/hpx/parallel/container_algorithms/partial_sort_copy.hpp [hunk 1]
```diff
@@ -307,14 +312,15 @@ namespace hpx { namespace ranges {
...
+    template <typename ExPolicy, typename Rng1, typename Rng2,
+        typename Compare = ranges::less,
+        typename Proj1 = parallel::util::projection_identity,
+        typename Proj2 = parallel::util::projection_identity>
     parallel::util::detail::algorithm_result_t<ExPolicy,
         partial_sort_copy_result<hpx::traits::range_iterator_t<Rng1>,
         hpx::traits::range_iterator_t<Rng2>>>
     partial_sort_copy(
-        ExPolicy&& policy, Rng1&& rng1, Rng2&& rng2, Comp&& comp = Comp(),
+        ExPolicy&& policy, Rng1&& rng1, Rng2&& rng2, Compare&& comp = Compare(),
```
Comment (dimitraka) [inline libs/core/algorithms/include/hpx/parallel/container_algorithms/partial_sort_copy.hpp:323]: Why do we use `Compare` here instead of `Comp`? I haven't seen it again so far

File: libs/core/algorithms/include/hpx/parallel/container_algorithms/partition.hpp [hunk 2]
```diff
@@ -134,56 +127,59 @@ namespace hpx { namespace ranges {
...
-        typename Pred, typename Proj>
-    typename util::detail::algorithm_result<ExPolicy,
-        subrange_t<FwdIter>>::type
-    partition(ExPolicy&& policy, FwdIter first, Sent last, Pred&& pred,
-        Proj&& proj);
+    template <typename ExPolicy, typename Rng, typename Pred,
+        typename Proj = parallel::util::projection_identity>
+    typename parallel::util::detail::algorithm_result<ExPolicy,
+        subrange_t<hpx::traits::range_iterator_t<Rng>>>::type
+    partition(ExPolicy&& policy, Rng&& rng, Pred&& pred, Proj&& proj = Proj());
```
Comment (dimitraka) [inline libs/core/algorithms/include/hpx/parallel/container_algorithms/partition.hpp:141]: Usually in the function declaration we have `Pred&& pred = Pred()` and not just `Pred&& pred`. Why is that here?

File: libs/core/algorithms/include/hpx/parallel/container_algorithms/partial_sort_copy.hpp (hunk 1 above)
Comment (hkaiser) [reply libs/core/algorithms/include/hpx/parallel/container_algorithms/partial_sort_copy.hpp:323]: It's just a name, but consistency matters. Please feel free to change it.

File: libs/core/algorithms/include/hpx/parallel/container_algorithms/partition.hpp (hunk 2 above)
Comment (hkaiser) [reply libs/core/algorithms/include/hpx/parallel/container_algorithms/partition.hpp:141]: Yes, ranges::partition requires a predicate (see https://eel.is/c++draft/alg.partitions#lib:is_partitioned)

File: libs/core/algorithms/include/hpx/parallel/container_algorithms/partition.hpp [hunk 3]
```diff
@@ -699,46 +702,52 @@ namespace hpx { namespace ranges {
...
-        partition_copy_result<FwdIter, OutIter2, OutIter3>>::type
-    partition_copy(ExPolicy&& policy, FwdIter1 first, Sent last,
-        FwdIter2 dest_true, FwdIter3 dest_false, Pred&& pred, Proj&& proj);
+    template <typename ExPolicy, typename Rng, typename FwdIter2,
+        typename FwdIter3, typename Pred,
+        typename Proj = parallel::util::projection_identity>
+    typename parallel::util::detail::algorithm_result<ExPolicy,
+        partition_copy_result<hpx::traits::range_iterator_t<Rng>, FwdIter2,
+            FwdIter3>>::type
+    partition_copy(ExPolicy&& policy, Rng&& rng, FwdIter2 dest_true,
```
[the remark this answers is not in the document]
Comment (hkaiser) [reply libs/core/algorithms/include/hpx/parallel/container_algorithms/partition.hpp:725]: The version taking an execution policy require for the iterators to be at least forward iterators (hence the name `FwdIter`). The sequential overload can be used with an output iterator, however. This kind of difference is true for many algorithms, just the docs do not reflect it properly.

File: libs/core/algorithms/include/hpx/parallel/container_algorithms/partial_sort_copy.hpp (hunk 1 above)
Comment (dimitraka) [reply libs/core/algorithms/include/hpx/parallel/container_algorithms/partial_sort_copy.hpp:323]: So, I will change `Compare&& comp = Compare()` to `Comp&& comp = Comp()` also in the code part. Is that ok?

**Comment by hkaiser** [comment]
@dimitraka just a heads-up: I have fixed the clang-format issues and have rebased this branch onto master.

**Comment by hkaiser** [comment]
bors merge
